edx_api/grades/__init__.py

```diff
@@ -0,0 +1,76 @@
+"""
+edX Grades REST API client class
+"""
+from requests.exceptions import HTTPError
+from six.moves.urllib.parse import urljoin  # pylint: disable=import-error
+
+from edx_api.enrollments import CourseEnrollments
+from .models import CurrentGrade, CurrentGrades
+
+
+class UserCurrentGrades(object):
+    """
+    edX student certificates client
+    """
+
+    def __init__(self, requester, base_url):
+        """
+        Args:
+            requester (Requester): an authenticated objects for requests to edX
+            base_url (str): string representing the base URL of an edX LMS instance
+        """
+        self.requester = requester
+        self.base_url = base_url
+
+    def get_student_current_grade(self, username, course_id):
+        """
+        Returns an CurrentGrade object for the user in a course
+
+        Args:
+            username (str): an edx user's username
+            course_id (str): an edX course id.
+
+        Returns:
+            CurrentGrade: object representing the student current grade for a course
+        """
+        # the request is done in behalf of the current logged in user
+        resp = self.requester.get(
+            urljoin(
+                self.base_url,
+                '/api/grades/v0/course_grade/{course_key}/users/?username={username}'.format(
+                    username=username,
+                    course_key=course_id
+                )
+            )
+        )
+
+        resp.raise_for_status()
+
+        return CurrentGrade(resp.json()[0])
+
+    def get_student_current_grades(self, username, course_ids=None):
```

Comment:
I do not think it's necessary: a `None` username will result in a request to edX containing the string "None" as username that will likely fail.